### PR TITLE
tidb_query: fix inconsistent error message between Coprocessor and TiDB

### PR DESCRIPTION
**Pull request by H-ZeX**
###  What have you changed?

- Change the way convert err to string. In the orignal way, the err's str like this `Eval("constant 18446744073709552000 overflows LongLong", 1690)`.
- make `cast a str that can not convert to decimal` return `0`, just like TiDB's cast
- cast xx to str according to whether it is unsigned and its bit size.

###  What is the type of the changes?

- Bugfix (a change which fixes an issue)

###  How is the PR tested?

- Unit test
- Integration test
- Manual test, the copr-test/pull/4 and copr-test/pull/3

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No.

###  Does this PR affect `tidb-ansible`?

No.

###  Refer to a related PR or issue link (optional)

Relate TiDB PR https://github.com/pingcap/tidb/pull/12858/

### CODE REVIEW

File: components/tidb_query/src/codec/mysql/decimal.rs [hunk 1]
```diff
@@ -1573,6 +1573,7 @@ impl Decimal {
         Decimal::from_bytes_with_word_buf(s, WORD_BUF_LEN)
     }
 
+    /// Return Err only whne the `s` is empty or can not convert to decimal
```
Comment (lonng) [inline components/tidb_query/src/codec/mysql/decimal.rs]:
```suggestion
    /// Returns error if the `s` is empty or can not be converted to decimal
```

File: components/tidb_query/src/codec/mysql/decimal.rs [hunk 2]
```diff
@@ -1811,7 +1812,13 @@ impl ConvertTo<Decimal> for &[u8] {
     //  TiDB's seems has bug, fix this after fix TiDB's
     #[inline]
     fn convert(&self, ctx: &mut EvalContext) -> Result<Decimal> {
-        let r = Decimal::from_bytes(self)?;
+        let r = match Decimal::from_bytes(self) {
+            Err(e) => {
+                debug!("Decimal::from_bytes failed, err is {:?}", e);
```
Comment (lonng) [inline components/tidb_query/src/codec/mysql/decimal.rs]: Should this part be merged to master?

File: components/tidb_query/src/codec/mysql/decimal.rs [hunk 3]
```diff
@@ -1811,7 +1825,7 @@ impl ConvertTo<Decimal> for &[u8] {
     //  TiDB's seems has bug, fix this after fix TiDB's
     #[inline]
     fn convert(&self, ctx: &mut EvalContext) -> Result<Decimal> {
-        let r = Decimal::from_bytes(self)?;
+        let r = Decimal::from_bytes(self).unwrap_or(Res::Ok(Decimal::zero()));
```
Comment (iosmanthus) [inline components/tidb_query/src/codec/mysql/decimal.rs]: Please change `unwrap_or` to `unwrap_or_else`, which make `clippy` happy.

**Pull request merged**